python/pyspark/sql/pandas/functions.pyi

```diff
@@ -46,25 +46,33 @@ class PandasUDFType:
     GROUPED_MAP: PandasGroupedMapUDFType
     GROUPED_AGG: PandasGroupedAggUDFType
 
+@overload
+def pandas_udf(f: PandasScalarToScalarFunction, returnType: Union[AtomicDataTypeOrString, ArrayType]) -> UserDefinedFunctionLike: ...  # type: ignore[misc]
```

Comment:
Can we add struct type and map types too (instead of Atomic types and array types)?

Reply:
Added `MapType`  for `PandasScalarUDFType` only (cause that `PandasScalarToStructFunction` part exists, no need to add `StructType`), added `MapType, StructType` for `PandasScalarIterUDFType`, added `MapType` for `PandasGroupedAggUDFType`(`StructType` is not supported), and for `PandasGroupedMapFunction`, no need to change.